From cf448f0bc88ea03ba8aea056779be6ab5a99c9aa Mon Sep 17 00:00:00 2001
From: Deve <deveee@gmail.com>
Date: Wed, 21 Apr 2021 11:33:59 +0200
Subject: [PATCH] Fixed a crash in settings in tts engine selection
 screen.

The crash happens when you try to uninstall current tts
engine and install a different one and then engine name
may be null.

If engine name is null then you can't select new engine
because of that crash, so that whole tts feature becomes
unusable.

Change-Id: Ie31a947eb6444242e8536db2dae1452ef74cc1ce
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 core/java/android/speech/tts/TtsEngines.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/speech/tts/TtsEngines.java b/core/java/android/speech/tts/TtsEngines.java
index ef63c8ff751d..3859418de217 100644
--- a/core/java/android/speech/tts/TtsEngines.java
+++ b/core/java/android/speech/tts/TtsEngines.java
@@ -498,7 +498,7 @@ public class TtsEngines {
      * specific preference in the list.
      */
     private static String parseEnginePrefFromList(String prefValue, String engineName) {
-        if (TextUtils.isEmpty(prefValue)) {
+        if (TextUtils.isEmpty(prefValue) || TextUtils.isEmpty(engineName)) {
             return null;
         }
 
-- 
2.43.0

